app/controllers/offers_controller.rb

```diff
@@ -16,8 +17,25 @@ def create
     end
   end
 
+  def destroy
+    @offer.destroy
+    redirect_to profile_path
+  end
+
+  def cancel
+    @offer.canceled!
+  end
+
+  def unignore
+    @offer.pending!
+  end
+
   private
 
+  def set_offer
+    @offer = Offer.find(params[:id])
```

Comment:
This means I can destroy/cancel/unignore any offer (not only my own). Need to bind it to current_user as usual